ci-operator/step-registry/optional-operators/kdump-configure-logs/optional-operators-kdump-configure-logs-commands.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+node_role=${APPLY_NODE_ROLE:=worker}
+log_path=${LOG_PATH:="/var/crash"}
+
+echo "Creating kdump configuration"
+kdump_conf=$(cat <<EOF | base64 -w 0
+path $log_path
+core_collector makedumpfile -l --message-level 7 -d 31
+EOF
+)
+
+echo "Creating kdump sysconfig"
+kdump_sysconfig=$(cat <<EOF | base64 -w 0
+KDUMP_COMMANDLINE_REMOVE="hugepages hugepagesz slub_debug quiet log_buf_len swiotlb"
+KDUMP_COMMANDLINE_APPEND="irqpoll nr_cpus=1 reset_devices cgroup_disable=memory mce=off numa=off udev.children-max=2 panic=10 rootflags=nofail acpi_no_memhotplug transparent_hugepage=never nokaslr novmcoredd hest_disable"
+KEXEC_ARGS="-s"
+KDUMP_IMG="vmlinuz"
+EOF
+)
+
+echo "Configuring kernel dumps on $node_role nodes"
+oc create -f - <<EOF
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: $node_role
+  name: 99-$node_role-kdump
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - contents:
+            source: data:text/plain;charset=utf-8;base64,${kdump_conf}
+          mode: 420
+          overwrite: true
+          path: /etc/kdump.conf
+        - contents:
+            source: data:text/plain;charset=utf-8;base64,${kdump_sysconfig}
+          mode: 420
+          overwrite: true
+          path: /etc/sysconfig/kdump
+    systemd:
+      units:
+        - enabled: true
+          name: kdump.service
+  kernelArguments:
+    - crashkernel=256M
+EOF
+
+echo "waiting for mcp/${node_role} condition=Updating timeout=5m"
+oc wait mcp/${node_role} --for condition=Updating --timeout=5m
+
+# When applying this configuration to the master node for single node
+# we need to wait for the master node to restart before we attempt to call out to the API Server
+if [[ "$node_role" == "master" ]]; then
```

Comment:
Might be best to add a single node topology check here as part of the if statement. I'm not 100% sure if `oc wait` will fail if the API server isn't responding, which will be the case during single node upgrade. So maybe something like this 
```
if node_role == master && single_node_topo; then 
wait 600seconds; 
else 
oc wait mcp/${node_role} --for condition=Updating --timeout=5m 
fi
```
 What do you think?

Reply:
Doesn't `node_role` == "master" essentially check if it's SNO? It would be "worker" for non-sno setups

Reply:
Not necessarily, `node_role` is identifying the the label we want to apply this machine config to, but you could have 3 master nodes, so `node_role` would still be `master` but not SNO.